.gitignore

```diff
@@ -96,20 +96,25 @@ nbgrader/docs/source/user_guide/source/ps1_autotest/problem1.html
 nbgrader/docs/source/user_guide/source/ps1_autotest/problem2.html
 
 # components stuff
-node_modules
-nbgrader/server_extensions/formgrader/static/components/bootstrap/less
-nbgrader/server_extensions/formgrader/static/components/bootstrap/js/*
-nbgrader/server_extensions/formgrader/static/components/datatables.net/js/jquery.dataTables.js
-nbgrader/server_extensions/formgrader/static/components/datatables.net-bs/js/dataTables.bootstrap.js
-nbgrader/server_extensions/formgrader/static/components/datatables.net-bs/css/dataTables.bootstrap.css
-nbgrader/server_extensions/formgrader/static/components/jquery/jquery-migrate.min.js
-nbgrader/server_extensions/formgrader/static/components/jquery/jquery-migrate.js
-nbgrader/server_extensions/formgrader/static/components/jquery-color/test
-nbgrader/server_extensions/formgrader/static/components/autosize/example
-
-!nbgrader/server_extensions/formgrader/static/components/autosize/dist
-!nbgrader/server_extensions/formgrader/static/components/bootstrap/js/bootstrap.js
-!nbgrader/server_extensions/formgrader/static/components/bootstrap/js/bootstrap.min.js
+/node_modules
+nbgrader/server_extensions/formgrader/static/node_modules/autosize/example
+nbgrader/server_extensions/formgrader/static/node_modules/autosize/src
+nbgrader/server_extensions/formgrader/static/node_modules/bootstrap/less
+nbgrader/server_extensions/formgrader/static/node_modules/bootstrap/js
+nbgrader/server_extensions/formgrader/static/node_modules/bootstrap/grunt
+nbgrader/server_extensions/formgrader/static/node_modules/jquery/external
+nbgrader/server_extensions/formgrader/static/node_modules/jquery/src
+nbgrader/server_extensions/formgrader/static/node_modules/underscore/amd
+nbgrader/server_extensions/formgrader/static/node_modules/underscore/cjs
+nbgrader/server_extensions/formgrader/static/node_modules/underscore/modules
+nbgrader/server_extensions/formgrader/static/node_modules/underscore/underscore-esm*
+nbgrader/server_extensions/formgrader/static/node_modules/underscore/underscore-node*
+nbgrader/server_extensions/formgrader/static/node_modules/underscore/underscore-umd*
```

Comment:
Because we're using the exceptions below, we can replace all of these specific rules with a more general one:

```
nbgrader/server_extensions/formgrader/static/node_modules/**/*
```

Reply:
Ah, I see what you mean. I think the change I want to see goes well beyond the scope of this PR. Thanks for clarifying.